### PR TITLE
Send aws_topology snapshot before raising errors

### PR DESCRIPTION
**Pull request by Eelviny**
### Step 1: Link to Jira issue
https://stackstate.atlassian.net/browse/STAC-13287

### Step 2: Description of changes
Ensure that `stop_snapshot` is called before raising all found errors, to ensure that one failing API does not block the sending of successful APIs.

### Step 3: Did you add / update tests for your changes in the right area?
- [x] Unit/Component test
- [ ] Integration test	

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: aws_topology/stackstate_checks/aws_topology/aws_topology.py [hunk 1]
```diff
@@ -142,14 +142,14 @@ def get_topology(self, instance_info, aws_client):
...
-                raise Exception("get_topology gave following exceptions: %s" % ", ".join(errors))
-        self.delete_ids += agent_proxy.delete_ids
+        # TODO this should be for tests, in production these relations should not be sent out
+        agent_proxy.finalize_account_topology()
+        self.components_seen = agent_proxy.components_seen
 
         self.stop_snapshot()
+        if len(errors) > 0:
+            raise Exception("get_topology gave following exceptions: %s" % ", ".join(errors))
+        self.delete_ids += agent_proxy.delete_ids
```
Comment (Jacco) [inline aws_topology/stackstate_checks/aws_topology/aws_topology.py:152]: Move this to under components_seen :-)

**Review by Jacco — CHANGES_REQUESTED** [review]

**Comment by Eelviny** [comment]
Continuing from the discussion in Slack -  @zandrewitte you had a concern that we're attempting to close a snapshot even if there are errors.

Yes, we are attempting to close if there are errors, but only the errors that we specifically catch (like the accessdenied errors). Any uncaught errors will make the snapshot collection completely fail as normal.

**Comment by Jacco** [comment]
Above statement is not how it is intended

**Comment by Eelviny** [comment]
This actually doesn't fix the underlying issue, closing

**Pull request closed without merging**